test/Microsoft.TestPlatform.Common.UnitTests/ExtensionFramework/Utilities/TestExtensionsTests.cs

```diff
@@ -239,4 +242,110 @@ public void GetExtensionsDiscoveredFromAssemblyShouldReturnTestDiscoveresAndLogg
+
+        // Make sure the method stays "pure" and returns a new reference regardless of the input.
+        merged1.Should().NotBeSameAs(first);
+        merged1.Should().NotBeSameAs(second);
+        merged2.Should().NotBeSameAs(first);
+        merged2.Should().NotBeSameAs(second);
+        merged1.Should().NotBeSameAs(merged2);
+    }
+
+    [TestMethod]
+    public void MergedDictionaryShouldBeEquivalentToTheExpectedDictionary()
+    {
+        var first = new Dictionary<string, HashSet<string>>
+        {
+            // Merged with "key1" from the next set.
+            { "key1", new HashSet<string>(new List<string>() { "ext1", "ext2", "ext3" }) },
+            // Empty hashset, will be removed from the result.
+            { "key2", new HashSet<string>() },
+            // Added as is.
+            { "key5", new HashSet<string>(new List<string>() { "ext1", "ext2" }) }
+        };
+        var second = new Dictionary<string, HashSet<string>>
+        {
+            // Merged with "key1" from the previous set.
+            { "key1", new HashSet<string>(new List<string>() { "ext2", "ext3", "ext3", "ext4", "ext5" }) },
+            // Empty hashset, will be removed from the result.
+            { "key2", new HashSet<string>() },
+            // Empty hashset, will be removed from the result.
+            { "key3", new HashSet<string>() },
+            // Added as is.
+            { "key4", new HashSet<string>(new List<string>() { "ext1" }) }
+        };
+        var expected = new Dictionary<string, HashSet<string>>
+        {
+            { "key1", new HashSet<string>(new List<string>() { "ext1", "ext2", "ext3", "ext4", "ext5" }) },
+            { "key4", new HashSet<string>(new List<string>() { "ext1" }) },
+            { "key5", new HashSet<string>(new List<string>() { "ext1", "ext2" }) }
+        };
+
+        // Merge the two dictionaries.
+        var merged = TestExtensions.CreateMergedDictionary(first, second);
+
+        // Make sure the merged dictionary has the exact same keys as the expected dictionary.
+        merged.Should().HaveCount(expected.Count);
+        merged.Should().ContainKeys(expected.Keys);
+        expected.Should().ContainKeys(merged.Keys);
+
+        // Make sure the hashsets for each key are equal.
+        merged.Values.Should().BeEquivalentTo(expected.Values);
+    }
+
+    [TestMethod]
+    public void AddExtensionTelemetryShouldAddJsonFormattedDiscoveredExtensionsTelemetry()
+    {
+        var telemetryData = new Dictionary<string, object>();
+        var extensions = new Dictionary<string, HashSet<string>>
+        {
+            { "key1", new HashSet<string>(new List<string>() { "ext1", "ext2", "ext3", "ext4", "ext5" }) },
+            { "key4", new HashSet<string>(new List<string>() { "ext1" }) },
+            { "key5", new HashSet<string>(new List<string>() { "ext1", "ext2" }) }
+        };
+
+        var expectedTelemetry =
+            @"{""key1"":[""ext1"",""ext2"",""ext3"",""ext4"",""ext5""],"
```

Comment:
Was thinking about https://docs.microsoft.com/dotnet/csharp/whats-new/csharp-11#raw-string-literals so something but current solution is better than all these backslashes :)

Reply:
So I've been experimenting with this new raw string functionality but I got stuck at removing line breaks from the multiline string. Since the JSON-formatted string we get by converting the extension dictionary to JSON has no additional spaces and no line brakes, that causes my assert to break. I can do a `Replace` but that seems overly complicated for a test and goes against a more "readable" string in my opinion, so it'll probably be best to stick with the current form of the string unless there's a simpler solution to remove line breaks that I may have missed.

For reference this is the way my raw string looks like right now:
```
var expectedTelemetry = """
    {"key1":["ext1","ext2","ext3","ext4","ext5"], <--LINE BREAK HERE
    "key4":["ext1"], <--LINE BREAK HERE
    "key5":["ext1","ext2"]}
    """;
```